### PR TITLE
NO-JIRA: use golang 1.22 image

### PR DESCRIPTION
**Pull request by Elbehery**
This PR update OCP/Etcd build image to use go1.22

cc @openshift/openshift-team-etcd

### CODE REVIEW

**Comment by tjungblu** [comment]
/lgtm

/hold

let's hold this for the first z release though

File: .ci-operator.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 build_root_image:
   name: release
   namespace: openshift
-  tag: rhel-9-release-golang-1.21-openshift-4.16
+  tag: rhel-9-release-golang-1.22-openshift-4.18
```
Comment (hasbro17) [inline .ci-operator.yaml]: Why is this using 4.18? Isn't this the 4.17 branch?

Comment (Elbehery) [reply .ci-operator.yaml]: Good catch !! 

 go1.22 image name was created for ocp18 today

Maybe Joep can help us here to be consistent ?

Comment (Elbehery) [reply .ci-operator.yaml]: @joepvd

Comment (Elbehery) [reply .ci-operator.yaml]: @hasbro17 updated 👍🏽

**Comment by Elbehery** [comment]
Hi @deads2k  

We need `jira/valid-bug` and `staff-eng-approved` here please 

/assign @deads2k

**Pull request merged**